### PR TITLE
Fixes scan-build warnings

### PR DESCRIPTION
**Pull request by huitema**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,10 +28,15 @@ before_script:
   - ./ci/build_picotls.sh
   - sudo sh -c 'echo 0 > /proc/sys/net/ipv6/conf/all/disable_ipv6'
 script:
-  - ls -l /usr/share/clang/scan-build
   # Now build picotls examples and test
-  - scan-build cmake -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=/usr/share/clang/scan-build/ccc-analyzer .
-  - scan-build make
+  - echo $CC
+  - echo $CXX
+  - if [ "$CC" == "clang" ]; then ls -l /usr/share/clang/scan-build; fi
```
Comment (alagoutte) [inline .travis.yml]: it is a test line... can be remove

File: .travis.yml [hunk 2]
```diff
@@ -28,10 +28,15 @@ before_script:
...
+  - if [ "$CC" == "clang" ]; then scan-build cmake -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=/usr/share/clang/scan-build/ccc-analyzer .; fi
+  - if [ "$CC" == "clang" ]; then scan-build make; fi
+  - cmake .;
```
Comment (alagoutte) [inline .travis.yml]: no need ; when there is no if

File: .travis.yml [hunk 3]
```diff
@@ -28,10 +28,15 @@ before_script:
...
+  - if [ "$CC" == "gcc" ]; then make clean; fi
+  - make;
```
Comment (alagoutte) [inline .travis.yml]: remove;

**Comment by huitema** [comment]
@alagoutte thanks for the review.  I am applying your changes and checking this in.

**Pull request merged**